src/benchmark/file_io_write_micro_benchmark.cpp

```diff
@@ -14,19 +14,48 @@ const int32_t MB = 1000000;
 
 class FileIOWriteMicroBenchmarkFixture : public MicroBenchmarkBasicFixture {
  public:
+  uint64_t control_sum = uint64_t{0};
+
   void SetUp(::benchmark::State& state) override {
     // TODO(phoeinx): Make setup/teardown global per file size to improve benchmark speed
     ssize_t BUFFER_SIZE_MB = state.range(0);
     // each int32_t contains four bytes
     int32_t vector_element_count = (BUFFER_SIZE_MB * MB) / 4;
     data_to_write = std::vector<int32_t>(vector_element_count, 42);
+    control_sum = vector_element_count * uint64_t{42};
 
     if (creat("file.txt", O_WRONLY) < 1) {
       std::cout << "create error" << std::endl;
     }
     chmod("file.txt", S_IRWXU);  // enables owner to rwx file
   }
 
+  void SanityCheck(int32_t NUMBER_OF_BYTES){
+    int32_t fd;
+    if ((fd = open("file.txt", O_RDONLY)) < 0) {
+      std::cout << "open error " << errno << std::endl;
+    }
+    chmod("file.txt", S_IRWXU);  // enables owner to rwx file
+    std::vector<uint32_t> read_data;
+    read_data.resize(NUMBER_OF_BYTES / 4);
+
+    // Getting the mapping to memory.
+    off_t OFFSET = 0;
+
+    int32_t* map = reinterpret_cast<int32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, OFFSET));
+    if (map == MAP_FAILED) {
+      std::cout << "Mapping Failed. " << std::strerror(errno) << std::endl;
+    }
+
+    memcpy(std::data(read_data), map, NUMBER_OF_BYTES);
+    auto sum = std::accumulate(read_data.begin(), read_data.end(), uint64_t{0});
```

Comment:
so we limit each element in the `read_data` vector to be of size `int32_t` and expect the sum to be expressable in `uint64_t`. With our current setup that should always work and I guess if we ever get overflow problems, we will find out and be able to debug it.

Reply:
The `read_data` elements should however also be `uint`s to avoid underflow errors